packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -218,6 +218,119 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
      *          RELAYER FUNCTIONS         *
      **************************************/
 
+    /**
+     * @notice Called by Relayer to execute a slow + fast relay from L2 to L1, fulfilling a corresponding deposit order.
+     * @dev There can only be one pending relay for a deposit. This method is effectively the relayDeposit and
+     * speedUpRelay methods concatenated. This could potentially be refactored to just call each method, but there
+     * are some gas savings in combining the transfers and hash computations.
+     * @dev Caller must have approved this contract to spend the total bond + amount - fees for `l1Token`.
+     * @param depositData the deposit data struct containing all the user's deposit information.
+     * @param realizedLpFeePct LP fee calculated off-chain considering the L1 pool liquidity at deposit time, before
+     *      quoteTimestamp. The OO acts to verify the correctness of this realized fee. Can not exceed 50%.
+     */
+    function relayAndSpeedUp(DepositData memory depositData, uint64 realizedLpFeePct) public nonReentrant() {
+        // If no pending relay for this deposit, then associate the caller's relay attempt with it.
+        uint32 priceRequestTime = uint32(getCurrentTime());
+
+        // The realizedLPFeePct should never be greater than 0.5e18 and the slow and instant relay fees should never be
+        // more than 0.25e18 each. Therefore, the sum of all fee types can never exceed 1e18 (or 100%).
+        require(
+            depositData.slowRelayFeePct < 0.25e18 &&
+                depositData.instantRelayFeePct < 0.25e18 &&
+                realizedLpFeePct < 0.5e18
+        );
+
+        // Check if there is a pending relay for this deposit.
+        bytes32 depositHash = _getDepositHash(depositData);
+
+        // Note: A disputed relay deletes the stored relay hash and enables this require statement to pass.
+        require(relays[depositHash] == bytes32(0), "Pending relay exists");
+
+        uint256 proposerBond = _getProposerBond(depositData.amount);
+        uint256 finalFee = store.computeFinalFee(address(l1Token)).rawValue;
+
+        // Save hash of new relay attempt parameters.
+        RelayData memory relayData =
+            RelayData({
+                relayState: RelayState.Pending,
+                slowRelayer: msg.sender, // Note: Increment numberOfRelays at the same time as setting relayId to its current value.
+                relayId: numberOfRelays++,
+                realizedLpFeePct: realizedLpFeePct,
+                priceRequestTime: priceRequestTime,
+                bond: proposerBond + finalFee
```

Comment:
One minor problem is that the OO decides the final fee at request time, so if the final fee changes between proposal and dispute, this could mean that the contract loses more or less of its balance than expected. Not sure how to get around this.

Reply:
Yeah this is an issue, perhaps you could reserve some X multiple of `bond` for now, that allows the OO final fee to change by +/-X%,  and returns to the proposer the excess after sending the final fee to the OO

Reply:
Or, we could just make an assumption that LPs take on the risk that final fee increases and they collectively pay the diff